File: Ejercicios Propuestos 1/Ejercicio 9/Ejercicio 9.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

# Escriba un programa que muestre la serie de Fibonacci para los valores comprendidos entre n y m.

# Solicitamos el límite inferior
n = int(input("Ingrese el valor de n (límite inferior): "))

# Solicitamos el límite superior
m = int(input("Ingrese el valor de m (límite superior): "))

# Inicializamos los dos primeros términos de Fibonacci
a = 0
b = 1

# Generamos la serie mientras no se supere el límite superior
while a <= m:

    # Verificamos si el término pertenece al intervalo [n,m]
    if a >= n:
        print(a)

    # Calculamos el siguiente término de la serie
    siguiente = a + b

    # Actualizamos los valores para continuar la secuencia
    a = b
    b = siguiente

File: Ejercicios Propuestos 1/Ejercicio 9/EJERCICIO_9.cpp
#include <iostream>

using namespace std;

int main() {
    int n, m, a, b, siguiente, temp_intercambio;

    cout << "Ingrese el límite inferior (n): ";
    cin >> n;
    cout << "Ingrese el límite superior (m): ";
    cin >> m;

    if (n > m) {
        temp_intercambio = n;
        n = m;
        m = temp_intercambio;
        cout << "Nota: Se acomodaron los límites automáticamente de menor a mayor." << endl;
    }

    a = 0;
    b = 1;

    cout << "Los números de Fibonacci entre " << n << " y " << m << " son:" << endl;

    while (a <= m) {
        if (a >= n) {
            cout << a << " ";
        }
        siguiente = a + b;
        a = b;
        b = siguiente;
    }

    cout << endl;

    return 0;
}
